pootle/apps/pootle_store/syncer.py

```diff
@@ -205,10 +205,10 @@ def get_new_units(self, old_ids, new_ids):
              in new_ids - old_ids])
 
     def get_obsolete_units(self, old_ids, new_ids):
-        return (
-            self.disk_store.findid(uid)
-            for uid
-            in old_ids - new_ids)
+        for uid in old_ids - new_ids:
+            unit = self.disk_store.findid(uid)
+            if unit and not unit.isobsolete():
```

Comment:
`not isobsolete`? The method is `get_obsolete_units`. Maybe I am missing something.